### DISCUSSION TITLE
Error new Table

### DISCUSSION THREAD

**Discussion by gustadelante** [fullstackhero/blazor-starter-kit#408] [Q&A] [answered]
Hi all.
I am a junior in c# and I am wanting to understand the architecture of Hero Blazor.
I'm testing a new entity with swagger and it looks like I need to configure something related to the mapper because it gives me the following error

<html>
<body>

Error:Response bodyDownload{   "Data": null,   "Messages": [     "Missing type map configuration or unsupported mapping.\r\n\r\nMapping types:\r\nObject -> GetTipoProductoByIdResponse\r\nSystem.Object -> ProdManager.Application.Features.TipoProducto.Queries.GetById.GetTipoProductoByIdResponse"   ],   "Succeeded": false }
--

</body>
</html>

The error appear in GetAllTipoProductoQuery.cs at the line: 

**var mappedTipoProducto = _mapper.Map<List<GetAllTipoProductoResponse>>(tipoProductoList);**

 public async Task<Result<List<GetAllTipoProductoResponse>>> Handle(GetAllTipoProductoQuery request, CancellationToken cancellationToken)
        {
            Func<Task<List<Domain.Entities.Catalog.TipoProducto>>> getAllTipoProducto = () => _unitOfWork.Repository<Domain.Entities.Catalog.TipoProducto>().GetAllAsync();
            var tipoProductoList = await _cache.GetOrAddAsync(ApplicationConstants.Cache.GetAllTipoProductoCacheKey, getAllTipoProducto);
            **var mappedTipoProducto = _mapper.Map<List<GetAllTipoProductoResponse>>(tipoProductoList);**
            return await Result<List<GetAllTipoProductoResponse>>.SuccessAsync(mappedTipoProducto);
        }
Please, if anyone gives me an idea of ​​what I am missing to configure, I will be very grateful.

Cheers!
Gus.

**Comment by gustadelante** [comment]
adding, i was create the Response:

namespace ProdManager.Application.Features.TipoProducto.Queries.GetAll
{
    public class GetAllTipoProductoResponse
    {
        public int Id { get; set; }
        public string Descripcion { get; set; }
        public string Estado { get; set; }
    }
}

**Comment by jeremiedevos** [comment, marked as answer]
Have you created a mapping profile? This is needed for automapper to know how to map it. 
If not navigate to Core/Application/Mappings and add a new class. 
For example:
```csharp
public class PriceProfile : Profile
{
    public PriceProfile()
    {
        CreateMap<AddEditPriceCommand, Price>().ReverseMap();
        CreateMap<GetPriceByIdResponse, Price>().ReverseMap();
        CreateMap<GetAllPricesResponse, Price>().ReverseMap();
    }
}
```
Not the inheritance from *Profile*, when the app is started, all the classes with this inheritance will be setup as a automapper profile. 
Hope this helped!

**Reply by gustadelante** [reply]
Hi Jeremi! 
Just! 

I was missing the mapping and couldn't find the problem... 

Thanks!
